universal-login-sdk/lib/sdk.ts

```diff
@@ -50,16 +51,14 @@ class UniversalLoginSDK {
     return [privateKey, contract.address];
   }
 
-  async getFutureWallet(onContractDeployed: OnContractDeployed, onBalanceChanged?: BalanceChangedCallback) {
+  async getFutureWallet() {
     await this.getRelayerConfig();
-    const [privateKey, contractAddress] = await createFutureWallet(this.config!.factoryAddress, this.provider);
-    this.getBalanceObserver();
-    this.balanceObserver!.startAndSubscribe(contractAddress, (tokenAddress, contractAddress) => this.onBalanceChanged(tokenAddress, contractAddress, onContractDeployed, onBalanceChanged));
-    return [privateKey, contractAddress];
+    this.getFutureWalletFactory();
+    return this.futureWalletFactory!.getFutureWallet();
   }
 
-  onBalanceChanged(tokenAddress: string, contractAddress: string, onContractDeployed: OnContractDeployed, onBalanceChanged?: BalanceChangedCallback) {
-    onBalanceChanged && onBalanceChanged(tokenAddress, contractAddress);
+  onReadyToDeploy(tokenAddress: string, contractAddress: string, onContractDeployed: OnContractDeployed, onReadyToDeploy?: ReadyToDeployCallback) {
```

Comment:
I don't think we need this method anymore, all should happen in `FutureWallet`.

Reply:
✔️